content/initiatives/codegen.md

```diff
@@ -0,0 +1,47 @@
++++
+title = "Making Rust faster than C"
+slug = "codegen"
+template = "initiatives/initiative.html"
+
+[extra]
+funders = [
+    "tweedegolf"
+]
+
+blogposts = [
+]
++++
+
+### Making Rust faster than C
+
+Rust, being a systems language, should be best in class for performance-critical software. While the performance difference with C is small, it currenly blocks Rust alternatives from being adopted.
+
+Our mission is to make critical infrastructure software safer. For that to happen in all relevant areas, performance considerations must to be removed as a blocking counter-argument.
+
+This initiative exists to contribute to change; to changing Rust's performance story with the ultimate goal of making it faster than C.
+
+### What we've done
+
+Our work on [zlib-rs](/initiatives/datacompression) led us on the path to improve the performance of state machines. Complex state machines are niche, but foundational to many programs, such as parsers, interpreters and networking protocols.
+
+Folkert de Vries wrote an [RFC](https://github.com/rust-lang/rfcs/pull/3720) to improve code generation for state machines and bjorn3 implemented a [proof of concept in the Rust compiler](https://github.com/trifectatechfoundation/rust/tree/labeled-match).
+
+The proof is in the eating: zlib-rs build using the PoC is **5% faster**.
```

Comment:
it's more than that in the relevant cases, ~14% to 15% for the small chunk sizes https://gist.github.com/folkertdev/977183fb706b7693863bd7f358578292

Reply:
it's just that for larger chunk sizes the difference is much smaller, because we enter a tight inner loop quickly